text/0224-experimental-attrribute-registration.md

```diff
@@ -0,0 +1,117 @@
+# Experimental Semantic Convention Attribute Registration
+
+A process for the registration of semantic convention attributes not yet included in a semantic convention.
+
+## Motivation
+
+Today, semantic conventions are "all or nothing."
+Each instrumentation implements one or more full semantic conventions with little room for experimentation.
+The lack of experimentation means there is often limited experience and evidence to guide  convention authors.
+It also often leaves instrumentation authors with little or no guidance if they wish to capture telemetry not yet covered by semantic conventions.
+This [Slack thread on #otel-specification](https://cloud-native.slack.com/archives/C01N7PP1THC/p1667997213067169?thread_ts=1667933561.313819&cid=C01N7PP1THC) highlights the problem in context of a specific semantic convention pull request.
+What should OpenTelemetry do with semantic convention proposals when there's unclear evidence that a particular signal is necessary or ubiquitous?
+
+## Explanation
+
+This proposal generally follows the recommendations of [RFC6648 section 4 “Recommendations for Protocol Designers.”](https://www.rfc-editor.org/rfc/rfc6648#section-4)
+RFC6648 recommends a registry of names be established which standardizes usage and prevents collisions.
+This OTEP separates a dictionary of semantic convention attributes from the semantic conventions, to be referenced by semantic conventions when necessary, and establishes a procedure for registering new and experimental semantic convention attributes.
+
+RFC6648 further recommends that X- prefix (or similar structure) MUST NOT be defined as experimental or nonstandard, and that names missing the X- prefix MUST NOT be defined as stable or standard.
+This is addressed below in [alternatives considered](#alternatives-considered).
+
+## Definitions
+
+**Attribute Dictionary**: The attribute dictionary contains all semantic convention attributes, including their names and value definitions.
```

Comment:
Currently, besides the name and value definitions, semantic conventions define a description, notes, examples, and requirement levels for attributes.

Is it intended that any of those are part of the dictionary? Or is this some enrichment that takes place in the semantic conventions?

Reply:
I think the attribute definition in the dictionary should cover the following:

- `name` (required)
- `type` (required)
- `description` (required)
- `examples` (required)
- `notes` (optional)
- `stability` (depending on what we decide on the above discussion)

`requirement_level`, `sampling relevance` and `tag` must not be part of the definition but will be specified on the usage of the attributes.

Reply:
Makes sense.

I think the OTEP should me more specific about this and give a definition of what the dictionary contains. It's important for understanding and working out how the attribute dictionary and semantic conventions interact, which I think is crucial.

Reply:
i'll update the otep to add more specificity

Reply:
I'm going to add this but I'm going to make it a "suggested" set of fields. If there is some concern during implementation that comes up I don't want the OTEP to disallow some flexibility to solve problems.